js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+        }
+        const request = {};
+        let coinId = undefined;
+        if (code !== undefined) {
+            coinId = coinDict[code];
+            request['coin_id'] = coinId;
+        }
+        const response = await this.privatePostApiV1CoinPrepayHistory (this.extend (request, params));
+        return this.parseTransactions (response['data']['list'], currency, since, limit);
+    }
+
+    async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const currency = undefined;
+        const coinRes = this.publicGetCoinAll ();
+        const coinList = this.safeValue (coinRes, 'data', []);
+        const coinDict = {};
+        const coinIdToCoinCode = {};
+        for (let i = 0; i < coinList.length; i++) {
+            coinDict[this.safeString (coinList[i], 'code', '')] = this.safeInteger (coinList[i], 'coin_id', 0);
+            coinIdToCoinCode[this.safeInteger (coinList[i], 'coin_id', 0)] = this.safeString (coinList[i], 'code', '');
+        }
+        const request = {};
+        let coinId = undefined;
+        if (code !== undefined) {
+            coinId = coinDict[code];
+            request['coin_id'] = coinId;
+        }
+        if (since !== undefined) {
+            request['startTime'] = since;
+        }
+        const response = await this.privatePostApiV1CoinWithdrawHistory (this.extend (request, params));
+        return this.parseTransactions (response['data']['list'], currency, since, limit);
+    }
+
+    parseTransactionStatusByType (status, type = undefined) {
+        if (type === undefined) {
+            return status;
+        }
+        const statuses = {
+            'deposit': {
+                '1': 'pending',
+                '2': 'entering',
+                '3': 'ok',
+                '4': 'failed',
+            },
+            'withdrawal': {
+                '1': 'pending',
+                '2': 'entering',
+                '3': 'ok',
+                '4': 'failed',
+                '5': 'not approved',
+            },
+        };
+        return (status in statuses[type]) ? statuses[type][status] : status;
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        const prepayId = this.safeString (transaction, 'prepay_id');
+        const withdrawId = this.safeString (transaction, 'withdraw_id');
+        const id = prepayId !== undefined ? prepayId : withdrawId;
+        const address = this.safeString (transaction, 'addr');
+        let tag = this.safeString (transaction, 'tag');
+        if (tag !== undefined) {
+            if (tag.length < 1) {
+                tag = undefined;
+            }
+        }
+        const txid = this.safeValue (transaction, 'tx_id');
+        const currencyId = this.safeString (transaction, 'coin_id');
+        const coinRes = this.publicGetCoinAll ();
```

Comment:
We cannot call async methods synchronously and we should not issue networking requests from within the parser. Need to remove this call from the parser in this line.

Reply:
Since the parameter required by our interface is coin_id, I need to obtain the code corresponding to coin_id, so that the corresponding code can be obtained through the dictionary by passing the code.

Reply:
@caoshitong369 see how `fetchCurrencies()` and `.currencies` work with all the other exchanges – those are loaded and cached once along with the markets as described here: https://github.com/ccxt/ccxt/wiki/Manual#loading-markets

Also see the following base methods:

JS:

- https://github.com/ccxt/ccxt/blob/master/js/base/Exchange.js#L658
- https://github.com/ccxt/ccxt/blob/master/js/base/Exchange.js#L617

Py:

- https://github.com/ccxt/ccxt/blob/master/python/ccxt/base/exchange.py#L1254
- https://github.com/ccxt/ccxt/blob/master/python/ccxt/base/exchange.py#L1213

PHP:

- https://github.com/ccxt/ccxt/blob/master/php/base/Exchange.php#L1515
- https://github.com/ccxt/ccxt/blob/master/php/base/Exchange.php#L1455

We should not do the networking requests in the parser, because that is a logical error, that code won't work, because it returns a promise that is not awaited in this case in async js/py.